src/target/riscv/debug_register_printers.c

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include <stdio.h>
+#include <inttypes.h>
+#include <assert.h>
+#include <stdarg.h>
+
+#include "debug_register_printers.h"
+
+static unsigned int get_len_or_sprintf(char *buf, unsigned int curr, const char *format, ...)
+{
+	assert(format);
+	va_list args;
+	int length;
+
+	va_start(args, format);
+	if (buf)
+		length = vsprintf(buf + curr, format, args);
+	else
+		length = vsnprintf(NULL, 0, format, args);
+	va_end(args);
+	assert(length >= 0);
+	return (unsigned int)length;
+}
+
+static unsigned int print_value(char *buf, unsigned int offset, uint64_t value)
```

Comment:
If you like, a suggestion for a more descriptive name:

```suggestion
static unsigned int print_number(char *buf, unsigned int offset, uint64_t value)
```

Reply:
Addressed. The corresponding change in `riscv-debug-spec`: https://github.com/riscv/riscv-debug-spec/pull/876